Assets/CrazyOptimizer/Editor/WindowComponents/ModelOptimizations/ModelOptimization.cs

```diff
@@ -0,0 +1,226 @@
+
+            EditorGUILayout.EndHorizontal();
+            EditorGUILayout.Space(5);
+
+            GUILayout.Label(
+                "This utility gives you an overview of the models used in your project. By optimizing various settings, you will be able to considerably decrease your final build size. You can click on a model to select it in the Project view. To find out more about how the tool finds the models, please check our GitHub repo.",
+                EditorStyles.wordWrappedLabel);
+
+            BuildExplanation("R/W enabled",
+                "When a Mesh is read/write enabled, Unity uploads the Mesh data to GPU-addressable memory, but also keeps it in CPU-addressable memory. In most cases, you should disable this option to save runtime memory usage.");
+            BuildExplanation("Polygons optimized",
+                "Optimize the order of polygons in the mesh to make better use of the GPUs internal caches to improve rendering performance.");
+            BuildExplanation("Vertices optimized",
+                "Optimize the order of vertices in the mesh to make better use of the GPUs internal caches to improve rendering performance.");
+            BuildExplanation("Mesh compression",
+                "Compressing meshes will decrease the final build size, but more compression introduces more artifacts in vertex data.");
+            BuildExplanation("Animation compression",
+                "Compressing animations will decrease the final build size, but more compression introduces more artifacts in the animations.");
+        }
+
+        static void BuildExplanation(string label, string explanation)
+        {
+            EditorGUILayout.BeginHorizontal();
+            GUILayout.Label(label, EditorStyles.boldLabel, GUILayout.Width(130));
+            GUILayout.Label(explanation, EditorStyles.wordWrappedLabel);
+            GUILayout.FlexibleSpace();
+            EditorGUILayout.EndHorizontal();
+        }
+
+        /**
+         * Find the paths of the scenes that will end up in the final build of the game.
+         */
+        static List<string> GetScenesInBuildPath()
+        {
+            var scenesInBuild = new List<string>();
+
+            foreach (var scene in EditorBuildSettings.scenes)
+            {
+                if (scene.enabled)
+                {
+                    scenesInBuild.Add(scene.path);
+                }
+            }
+
+            return scenesInBuild;
+        }
+
+        /**
+         * Find recursively the models on which this scene depends.
+         */
+        static List<string> GetSceneModelDependencies(string scenePath)
+        {
+            var modelDependencies = new List<string>();
+            var assetDependencies = AssetDatabase.GetDependencies(scenePath, true);
+
+            foreach (var assetDependency in assetDependencies)
+            {
+                if (AssetDatabase.GetMainAssetTypeAtPath(assetDependency) == typeof(GameObject))
```

Comment:
You are right. I just checked and prefabs for example also equal to a `GameObject` and this would result in a lot of unnecessary warning logs. I think an extra check for file extension could help based on the documentation of the supported model file formats: [https://docs.unity3d.com/2020.1/Documentation/Manual/3D-formats.html](https://docs.unity3d.com/2020.1/Documentation/Manual/3D-formats.html)

```
static List<string> _modelFormats = new List<string>() { ".fbx", ".dae", ".3ds", ".dxf", ".obj" };

static bool IsModel(string assetDependency)
{
    return AssetDatabase.GetMainAssetTypeAtPath(assetDependency) == typeof(GameObject) &&
        _modelFormats.Contains(System.IO.Path.GetExtension(assetDependency).ToLowerInvariant());
}
```

What do you think?

Reply:
Yep, looks nice.

There is also the option of adding native file formats supported by Unity, fx .blend, but since the guys from Unity themselves don't recommend using native files, I'd say we skip them.

Reply:
I added this extra check in my last commit: [https://github.com/CrazyGamesCom/unity-optimizations-package/pull/6/commits/7a231338ad417ec649bbbb8468fc5caa9661917d](https://github.com/CrazyGamesCom/unity-optimizations-package/pull/6/commits/7a231338ad417ec649bbbb8468fc5caa9661917d)